File: leetcode/2medium/위성원/1362.ts
function closestDivisors(num: number): number[] {
  const num1 = num + 1;
  const num2 = num + 2;

  let min = Infinity;
  let answer: number[] = [];

  let i = 1;
  let j = num1;
  while (Math.sqrt(j) >= i) {
    if (num1 % i === 0) {
      const d = num1 / i;
      if (min > Math.abs(d - i)) {
        min = Math.abs(d - i);
        answer = [d, i];
      }
    }
    i++;
  }

  i = 1;
  j = num2;
  while (Math.sqrt(j) >= i) {
    if (num2 % i === 0) {
      const d = num2 / i;
      if (min > Math.abs(d - i)) {
        min = Math.abs(d - i);
        answer = [d, i];
      }
    }
    i++;
  }

  return answer;
}

File: leetcode/2medium/이경호/1362.cpp
class Solution {
public:
  vector<int> closestDivisors(int num) {
    int num1 = num + 1;
    int num2 = num + 2;
    vector<int> num1s;
    vector<int> num2s;
    for (int i = ceil(sqrt(num1)); i > 0; i--) {
      if (num1 % i == 0) {
        num1s.push_back(num1 / i);
        num1s.push_back(i);
        break;
      }
    }
    for (int i = ceil(sqrt(num2)); i > 0; i--) {
      if (num2 % i == 0) {
        num2s.push_back(num2 / i);
        num2s.push_back(i);
        break;
      }
    }
    if (num1s[0] - num1s[1] < num2s[0] - num2s[1]) {
      return num1s;
    }
    else {
      return num2s;
    }
  }
};